pallets/loans/src/migrations.rs

```diff
@@ -0,0 +1,89 @@
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::*;
+
+/// This migration nukes all storages from the pallet individually.
+pub struct Migration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for Migration<T> {
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+		// The current state from altair when this migration should be done has only one element
+		// in each store.
+		ensure!(
+			PoolToLoanNftClass::<T>::iter_values().count() == 1,
+			"Err PoolToLoanNftClass"
+		);
+		ensure!(
+			LoanNftClassToPool::<T>::iter_values().count() == 1,
+			"Err LoanNftClassToPool"
+		);
+		ensure!(
+			NextLoanId::<T>::iter_values().count() == 1,
+			"Err NextLoanId"
+		);
+		ensure!(Loan::<T>::iter_values().count() == 1, "Err Loan");
+		ensure!(
+			ActiveLoans::<T>::iter_values().count() == 1,
+			"Err ActiveLoans"
+		);
+		ensure!(
+			ClosedLoans::<T>::iter_values().count() == 1,
+			"Err ClosedLoans"
+		);
+		ensure!(PoolNAV::<T>::iter_values().count() == 1, "Err PoolNAV");
+		ensure!(
+			PoolWriteOffGroups::<T>::iter_values().count() == 1,
+			"Err PoolWriteOffGroups "
+		);
+
+		Ok(Vec::new())
+	}
+
+	fn on_runtime_upgrade() -> Weight {
+		let _ = PoolToLoanNftClass::<T>::clear(1, None);
+		let _ = LoanNftClassToPool::<T>::clear(1, None);
+		let _ = NextLoanId::<T>::clear(1, None);
+		let _ = Loan::<T>::clear(1, None);
+		let _ = ActiveLoans::<T>::clear(1, None);
+		let _ = ClosedLoans::<T>::clear(1, None);
+		let _ = PoolNAV::<T>::clear(1, None);
+		let _ = PoolWriteOffGroups::<T>::clear(1, None);
+
+		T::DbWeight::get().writes(8)
+	}
```

Comment:
I was a bit baffled why the `try-runtime` logs did not show any migration for `Loans` so I added logs to the `pre` and `post` hooks as well as the migration itself which were executed as expected. 

So we could add some assurance log at least to `on_runtime_upgrade`, better to both hooks as well.

```rust
log::debug!("Loans storage clearing migration successful");
```

Reply:
Ok, I agree!